### PR TITLE
Feature/package and build fixes

### LINKED ISSUE #66: Fix iconUrl and licenseUrl in nuget packages
- Cake.Frosting
  - [ ] icon
- Cake.Frosting.Template
  - [ ] licence
  - [ ] icon
 
To address:
```diff
- WARNING: The <iconUrl> element is deprecated. 
- Consider using the <icon> element instead. https://aka.ms/deprecateIconUrl
- WARNING: The <licenseUrl> element is deprecated.
- Consider using the <license> element instead. Learn more: https://aka.ms/deprecateLicenseUrl.
```

### PR DESCRIPTION
**Pull request by Roemer**
I'd love to give some life back to this project as I see this as a much better way which is also similar to NUKE's concept which allows much better tooling support from VS/VSCode.

This is a first change to allow easier building by allowing newer .NET Core SDKS and fixes the warnings for the packages.
Also the build project is added to the main solution so only one solution needs to be opened.

Fixes #66

### CODE REVIEW

**Comment by Roemer** [comment]
The build fails as the image used in appveyor seems to be quite old and uses a nuget.exe which does not support the `license` tag. Can someone take care of this?

**Comment by devlead** [comment]
Change AppVeyor image should be addressable in this PR updating [appveyor.yml](https://github.com/cake-build/frosting/blob/develop/appveyor.yml)

https://www.appveyor.com/docs/build-environment/#choosing-image-for-your-builds

File: template/Cake.Frosting.Template.nuspec [hunk 1]
```diff
@@ -5,8 +5,9 @@
         <description>Cake.Frosting templates for the .NET SDK.</description>
         <summary>Cake (C# Make) is a build automation system with a C# DSL to do things like compiling code, copy files/folders, running unit tests, compress files and build NuGet packages.</summary>
         <authors>Patrik Svensson, Mattias Karlsson, Gary Ewan Park, Alistair Chapman, Martin Björkström and contributors</authors>
-        <licenseUrl>https://github.com/cake-build/frosting/blob/develop/LICENSE</licenseUrl>
+        <license type="expression">MIT</license>
         <projectUrl>https://github.com/cake-build/frosting</projectUrl>
+        <icon>cake-medium.png</icon>
```
Comment (devlead) [inline template/Cake.Frosting.Template.nuspec:10]: Image would need to be added to repo to be accessible from CI.

File: src/Cake.Frosting/Cake.Frosting.csproj [hunk 2]
```diff
@@ -5,6 +5,7 @@
     <Copyright>Copyright (c) .NET Foundation and contributors</Copyright>
     <Authors>Patrik Svensson, Mattias Karlsson, Gary Ewan Park, Alistair Chapman, Martin Björkström and contributors</Authors>
     <PackageTags>Cake;Build;Build automation</PackageTags>
+    <PackageIcon>cake-medium.png</PackageIcon>
     <PackageIconUrl>https://raw.githubusercontent.com/cake-build/graphics/master/png/cake-medium.png</PackageIconUrl>
```
Comment (devlead) [inline src/Cake.Frosting/Cake.Frosting.csproj:9]: PackageIconUrl and PackageIcon shouldn't be combined it's either or.

File: build/global.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "sdk": {
-    "version": "2.1.802"
+    "version": "2.1.803",
```
Comment (devlead) [inline build/global.json:3]: Beleive we need 3.x SDK (think it came in 3.0.100) so we should probably update to latest LTS SDK which is 3.1.301.

File: template/Cake.Frosting.Template.nuspec (hunk 1 above)
Comment (gep13) [reply template/Cake.Frosting.Template.nuspec:10]: Looks to me like this was added above, right?

Comment (devlead) [reply template/Cake.Frosting.Template.nuspec:10]: Ah correct my mistake.

**Comment by devlead** [comment]
@Roemer your changes have been merged, thanks for your contribution 👍

File: src/Cake.Frosting/Cake.Frosting.csproj (hunk 2 above)
Comment (Roemer) [reply src/Cake.Frosting/Cake.Frosting.csproj]: According to the tip in https://docs.microsoft.com/en-us/nuget/reference/nuspec which states `You can specify both icon and iconUrl to maintain backward compatibility with sources that do not support icon.`, I left the Url in. But it's fine for me to remove it.

Comment (devlead) [reply src/Cake.Frosting/Cake.Frosting.csproj]: Think we need to update the SDK for it to work properly
Icon is added but `PackageIcon` ignored dotnet by pack so it doesn't end up in the final nuspec

Comment (devlead) [reply src/Cake.Frosting/Cake.Frosting.csproj]: Raised issue #73 for that.

Comment (Roemer) [reply src/Cake.Frosting/Cake.Frosting.csproj]: Makes probably sense to anyway go to the latest LTS.

Comment (devlead) [reply src/Cake.Frosting/Cake.Frosting.csproj]: Agreed.

**Pull request merged**